jailer/src/lib.rs

```diff
@@ -97,6 +96,122 @@ pub enum Error {
+            MissingArgument(ref arg) => write!(f, "Missing argument: {}", arg),
+            MissingParent(ref path) => write!(f, "File {:?} doesn't have a parent", path),
+            MkdirOldRoot(ref err) => write!(
+                f,
+                "Failed to create the jail root directory before pivoting root: {}",
+                err
+            ),
+            MknodDevNetTun(ref err) => write!(
+                f,
+                "Failed to create /dev/net/tun via mknod inside the jail: {}",
+                err
+            ),
+            MountBind(ref err) => {
+                write!(f, "Failed to bind mount the jail root directory: {}", err)
+            }
+            MountPropagationPrivate(ref err) => write!(
+                f,
+                "Failed to change the propagation type to private: {}",
+                err
+            ),
+            NotAFile(ref path) => write!(f, "{:?} is not a file", path),
+            NumaNode(ref node) => write!(f, "Invalid numa node: {}", node),
+            OpenDevKvm(ref err) => write!(f, "Failed to open /dev/kvm: {}", err),
+            OpenDevNull(ref err) => write!(f, "Failed to open /dev/null: {}", err),
+            OsStringParsing(ref path, _) => {
+                write!(f, "Failed to parse path {:?} into an OsString", path)
+            }
+            PivotRoot(ref err) => write!(f, "Failed to pivot root: {}", err),
+            ReadLine(ref path, ref err) => {
+                write!(f, "Failed to read line from {:?}: error {:?}", path, err)
+            }
+            ReadToString(ref path, ref err) => write!(
+                f,
+                "Failed to read file {:?} into a string: error {:?}",
+                path, err
+            ),
+            RegEx(ref err) => write!(f, "Regex failed: {:?}", err),
+            RmOldRootDir(ref err) => write!(f, "Failed to remove old jail root directory: {}", err),
+            SeccompLevel(ref err) => write!(f, "Failed to parse seccomp level: error {:?}", err),
+            SetCurrentDir(ref err) => {
+                write!(f, "Failed to change current directory: error {:?}", err)
+            }
+            SetNetNs(ref err) => write!(f, "Failed to join network namespace: netns: {}", err),
+            SetSid(ref err) => write!(f, "Failed to daemonize: setsid: {}", err),
+            Uid(ref uid) => write!(f, "Invalid uid: {}", uid),
+            UmountOldRoot(ref err) => write!(f, "Failed to unmount the old jail root: {}", err),
+            UnexpectedKvmFd(fd) => write!(f, "Unexpected value for the /dev/kvm fd: {}", fd),
+            UnexpectedListenerFd(fd) => {
+                write!(f, "Unexpected value for the socket listener fd: {}", fd)
+            }
+            UnshareNewNs(ref err) => {
+                write!(f, "Failed to unshare into new mount namespace: {}", err)
+            }
+            UnixListener(ref err) => {
+                write!(f, "Failed to bind to the Unix socket: error {:?}", err)
```

Comment:
```suggestion
                write!(f, "Failed to bind to the Unix socket: {}", err)
```

Reply:
The inner error is an `io::Error` : `UnixListener(io::Error)` - which doesn't implement `Display`, so formatting with `{}` won't work.